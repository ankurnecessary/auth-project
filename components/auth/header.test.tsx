import { render, screen } from '@testing-library/react';
import Header from '@/components/auth/header'; // Adjust path accordingly
import '@testing-library/jest-dom';
import { Poppins } from 'next/font/google';

describe('Header component', () => {
  const label = 'Welcome to the authentication page';

  it('renders the header with the correct label', () => {
    render(<Header label={label} />);

    // Check if the label text is rendered
    const labelElement = screen.getByText(label);
    expect(labelElement).toBeInTheDocument();
    expect(labelElement).toHaveTextContent(label);
  });

  it('renders the h1 element with correct text and icon', () => {
    render(<Header label={label} />);

    // Check if the h1 contains the correct text "ðŸ”‘ Auth"
    const headingElement = screen.getByRole('heading', { level: 1 });
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toHaveTextContent('ðŸ”‘ Auth');
  });

  it('applies the correct Poppins font class', () => {
    const { container } = render(<Header label={label} />);

    // Get the h1 element and check if it has the correct font class
    const headingElement = container.querySelector('h1');
    const fontClassName = Poppins({
      subsets: ['latin'],
      weight: ['600'],
    }).className;

    expect(headingElement).toHaveClass(fontClassName);
  });

  it('applies the correct class names to the header elements', () => {
    render(<Header label={label} />);

    // Check if the h1 element has the correct classes
    const headingElement = screen.getByRole('heading', { level: 1 });
    expect(headingElement).toHaveClass('text-3xl font-semibold');

    // Check if the paragraph element has the correct classes
    const paragraphElement = screen.getByText(label);
    expect(paragraphElement).toHaveClass('text-sm text-muted-foreground');
  });
});
