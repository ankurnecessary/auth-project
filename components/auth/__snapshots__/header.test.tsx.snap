// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot when label is provided 1`] = `
<DocumentFragment>
  <div
    class="flex w-full flex-col items-center justify-center gap-y-4"
  >
    <h1
      class="text-3xl font-semibold className"
    >
      ðŸ”‘ Auth
    </h1>
    <p
      class="text-sm text-muted-foreground"
    >
      Welcome to the authentication page
    </p>
  </div>
</DocumentFragment>
`;
